src/main/java/hanghaeboard/api/exception/ApiControllerAdvice.java

```diff
@@ -0,0 +1,52 @@
+package hanghaeboard.api.exception;
+
+import hanghaeboard.api.ApiResponse;
+import hanghaeboard.api.exception.exception.InvalidPasswordException;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindException;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class ApiControllerAdvice {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ApiResponse<Object> handleValidationExceptions(MethodArgumentNotValidException ex) {
+        ObjectError objectError = ex.getAllErrors().get(0);
+
+        return ApiResponse.of(HttpStatus.BAD_REQUEST, objectError.getDefaultMessage());
+    }
+
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(BindException.class)
+    public ApiResponse<Object> bindException(BindException e) {
+        ObjectError objectError = e.getAllErrors().get(0);
+
+        return ApiResponse.of(HttpStatus.BAD_REQUEST, objectError.getDefaultMessage());
+    }
+
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ApiResponse<Object> entityNotFoundExceptions(EntityNotFoundException e) {
+        return ApiResponse.of(HttpStatus.NOT_FOUND, e.getMessage());
+    }
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(InvalidPasswordException.class)
+    public ApiResponse<Object> invalidPasswordExceptions(InvalidPasswordException e) {
+        return ApiResponse.of(HttpStatus.BAD_REQUEST, e.getMessage());
+    }
+
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler(Exception.class)
+    public ApiResponse<Object> exception(Exception e){
+        return ApiResponse.of(HttpStatus.INTERNAL_SERVER_ERROR, "서버에 예상치 못한 오류가 발생했습니다.");
```

Comment:
```suggestion
        log.error("Internal error occured", e);
        return ApiResponse.of(HttpStatus.INTERNAL_SERVER_ERROR, "서버에 예상치 못한 오류가 발생했습니다.");
```

로깅을 안하면 서버에 어떤 에러가 발생했는지 트래킹하기 어려우니 로깅을 남겨두는 것도 좋아보입니다!